frang/test_http_resp_code_block.py

```diff
@@ -0,0 +1,62 @@
+"""Functional tests for http_resp_code_block."""
+
+import os
+
+from framework import tester, deproxy_server
+from helpers import tf_cfg, deproxy, tempesta, control
+from framework.templates import fill_template
+
+__author__ = 'Tempesta Technologies, Inc.'
+__copyright__ = 'Copyright (C) 2019 Tempesta Technologies, Inc.'
+__license__ = 'GPL2'
+
+class HttpRespCodeBlock(tester.TempestaTest):
+    backends = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'port' : '8000',
+            'response' : 'static',
+            'response_content' : """HTTP/1.1 404 OK
+Content-Length: 0
+
+"""
+        },
+    ]
+
+    tempesta = {
+        'config' : """
+server ${general_ip}:8000;
+
+frang_limits {
+    http_resp_code_block 404 20 5;
+}
+""",
+    }
+
+    clients = [
+        {
+            'id' : 'deproxy',
+            'type' : 'deproxy',
+            'addr' : "${tempesta_ip}",
+            'port' : '80'
+        },
+    ]
+
+    def test(self):
```

Comment:
Only one test case is checked: when a client generates a bunch of requests in one single moment. There are more situations to cover:
- client constantly generate requests, some of them are responded by 200 code, some - 404. But there is very low rate of 404 responses, so client is not blocked dirung. Test duration must be atleast 5*2 seconds
- Same as above, but 404 rate is slightly more than allowed, client must be blocked. No bursting! we need validate that 21st 404-response in 5-second time frame will case connection close.

The limit is per-connection limit, so we must  use several connection in the test: good and bad. only bad connections must be closed.